.gitignore

```
@@ -6,6 +6,7 @@
 *.a
 *-opt
 *-dbg
+*-debug
```

where does this come from?

---

it s missing. You could be checking in your debug executable by accident currently

sorry for the rider commit, I can boot it to a fixup PR if you prefer

---

the debug executable is `dbg` which is already in the `.gitignore`

---

```suggestion
```

---

You can compile with METHOD=debug and it works too
``
(moose) giudgl@fn654105:~/projects/moose_v2/framework$ ls
Makefile               build_support          include                libmoose-devel.0.dylib notes.md
app.mk                 contrib                libmoose-debug.0.dylib libmoose-devel.dylib   plugin.mk
build                  data                   libmoose-debug.dylib   libmoose-devel.la      scripts
build.mk               doc                    libmoose-debug.la      moose.mk               src
(moose) giudgl@fn654105:~/projects/moose_v2/framework$ ldd libmoose-debug.dylib 
libmoose-debug.dylib:
	/Users/giudgl/projects/moose_v2/framework/libmoose-debug.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/Users/giudgl/projects/moose_v2/framework/contrib/pcre/libpcre-debug.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/../libmesh/lib/libmesh_dbg.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/../libmesh/lib/libtimpi_dbg.11.dylib (compatibility version 12.0.0, current version 12.0.0)
	@rpath/libhdf5_cpp.200.dylib (compatibility version 202.0.0, current version 202.0.0)
	@rpath/libvtkIOCore-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkCommonCore-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkCommonDataModel-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkFiltersCore-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkIOXML-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkImagingCore-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkIOImage-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkImagingMath-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkIOParallelXML-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkParallelMPI-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkParallelCore-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libvtkCommonExecutionModel-9.1.1.dylib (compatibility version 1.0.0, current version 9.1.0)
	@rpath/libz.1.dylib (compatibility version 1.0.0, current version 1.2.12)
	@rpath/libslepc.3.16.dylib (compatibility version 3.16.0, current version 3.16.2)
	@rpath/libpetsc.3.16.dylib (compatibility version 3.16.0, current version 3.16.6)
	@rpath/libHYPRE-2.23.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libstrumpack.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libsuperlu_dist.7.dylib (compatibility version 7.0.0, current version 7.1.1)
	@rpath/libhdf5_hl.200.dylib (compatibility version 201.0.0, current version 201.1.0)
	@rpath/libhdf5.200.dylib (compatibility version 202.0.0, current version 202.0.0)
	@rpath/libparmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libmpifort.12.dylib (compatibility version 15.0.0, current version 15.2.0)
	@rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
	@rpath/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	@rpath/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
	@rpath/libmpicxx.12.dylib (compatibility version 15.0.0, current version 15.2.0)
	@rpath/libomp.dylib (compatibility version 5.0.0, current version 5.0.0)
``

---

```
(moose) giudgl@fn654105:~/projects/moose_v2/modules/navier_stokes$ ll
total 672
drwxr-xr-x  17 giudgl  INEL-NT\Domain Users     544 Oct 13 17:59 ./
drwxr-xr-x  34 giudgl  INEL-NT\Domain Users    1088 Oct 12 19:06 ../
drwxr-xr-x   2 giudgl  INEL-NT\Domain Users      64 Oct 13 17:59 .libs/
-rw-r--r--   1 giudgl  INEL-NT\Domain Users  198997 Oct 13 17:59 .previous_test_results.json
-rw-r--r--   1 giudgl  INEL-NT\Domain Users    1077 Sep 20 08:30 Makefile
drwxr-xr-x   4 giudgl  INEL-NT\Domain Users     128 Oct 13 17:58 build/
drwxr-xr-x  10 giudgl  INEL-NT\Domain Users     320 Sep 27 22:51 doc/
drwxr-xr-x   4 giudgl  INEL-NT\Domain Users     128 Sep 27 22:51 examples/
drwxr-xr-x  18 giudgl  INEL-NT\Domain Users     576 Sep 27 22:51 include/
drwxr-xr-x   6 giudgl  INEL-NT\Domain Users     192 Oct 13 17:59 lib/
-rwxr-xr-x   1 giudgl  INEL-NT\Domain Users  132960 Oct 13 17:59 navier_stokes-debug*
drwxr-xr-x   3 giudgl  INEL-NT\Domain Users      96 Sep 20 08:30 python/
lrwxr-xr-x   1 giudgl  INEL-NT\Domain Users      23 Sep 20 08:30 run_tests@ -> ../../scripts/run_tests
drwxr-xr-x  22 giudgl  INEL-NT\Domain Users     704 Oct 13 17:59 src/
drwxr-xr-x   6 giudgl  INEL-NT\Domain Users     192 Oct 13 17:59 test/
-rw-r--r--   1 giudgl  INEL-NT\Domain Users      92 Sep 20 08:30 testroot
drwxr-xr-x   5 giudgl  INEL-NT\Domain Users     160 Sep 27 22:51 unit/
(moose) giudgl@fn654105:~/projects/moose_v2/modules/navier_stokes$ METHOD=debug ./run_tests 
test:finite_element/ins/RZ_cone.jac ....................................................... [METHOD!=OPT] SKIP
test:finite_element/ins/RZ_cone.ad_rz_cone_by_parts_traction_steady_supg_pspg_jac ......... [METHOD!=OPT] SKIP
test:finite_element/ins/RZ_cone.ad_jac .................................................... [METHOD!=OPT] SKIP
test:finite_volume/ins/channel-flow.triangles/tets .............................................. [HEAVY] SKIP
test:finite_volume/ins/channel-flow.mixing-length-jac ..................................... [METHOD!=OPT] SKIP
test:finite_element/ins/lid_driven.ad_lid_driven_action ................ [METHOD!=OPT,skipped dependency] SKIP
test:finite_element/ins/lid_driven.ad_lid_driven_mean_zero_pressure ....................... [METHOD!=OPT] SKIP
test:finite_element/ins/lid_driven.still_unstable ......................................... [METHOD!=OPT] SKIP
test:finite_element/ins/lid_driven.ad_lid_driven .......................................... [METHOD!=OPT] SKIP
test:finite_element/ins/lid_driven.ad_lid_driven_stabilized_jacobian ...................... [METHOD!=OPT] SKIP
test:finite_element/ins/lid_driven.ad_lid_driven_jacobian ................................. [METHOD!=OPT] SKIP
test:finite_element/ins/coupled-force.steady-function-jac ................................. [METHOD!=OPT] SKIP
test:finite_element/ins/coupled-force.steady-action-jac ................................... [METHOD!=OPT] SKIP
test:finite_element/ins/coupled-force.steady-action-function-jac .......................... [METHOD!=OPT] SKIP
test:finite_element/ins/coupled-force.steady-jac .......................................... [METHOD!=OPT] SKIP
test:finite_element/ins/mms/supg.alpha_1e-3/4x4 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/supg.alpha_1e-6/8x8 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/supg.alpha_1e-3/16x16 ............................................... [HEAVY] SKIP
test:finite_element/ins/mms/supg.alpha_1e-6/32x32 ............................................... [HEAVY] SKIP
test:finite_element/ins/mms/supg.alpha_1e-6/4x4 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/supg.alpha_1e-3/8x8 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/supg.alpha_1e-6/16x16 ............................................... [HEAVY] SKIP
test:finite_element/ins/mms/supg.alpha_1e-3/32x32 ............................................... [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-3/4x4 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-6/8x8 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-3/32x32 ............................................... [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-6/32x32 ............................................... [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-3/16x16 ............................................... [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-6/4x4 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-3/8x8 ................................................. [HEAVY] SKIP
test:finite_element/ins/mms/pspg.alpha_1e-6/16x16 ............................................... [HEAVY] SKIP
test:finite_volume/ins/lid-driven.jacobian ................................................ [METHOD!=OPT] SKIP
test:finite_volume/ins/lid-driven.quiescent_jac ........................................... [METHOD!=OPT] SKIP
test:finite_volume/ins/mms/channel-flow.plane-poiseuille-rc-first ............................... [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow.plane-poiseuille-average-first .......................... [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow.2d-rc ................................................... [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow.2d-rc-with-temp ......................................... [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow.plane-poiseuille-rc ..................................... [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow.2d-average-with-temp .................................... [HEAVY] SKIP
test:finite_element/ins/boussinesq.jac .................................................... [METHOD!=OPT] SKIP
test:finite_volume/wcns/channel-flow.transient-jac ........................................ [METHOD!=OPT] SKIP
test:finite_volume/wcns/channel-flow.turbulence-jac ....................................... [METHOD!=OPT] SKIP
test:finite_volume/pins/mms/porosity_change.pressure-corrected-jac ........................ [METHOD!=OPT] SKIP
test:finite_element/ins/boussinesq/benchmark.1e5 ................................................ [HEAVY] SKIP
test:finite_element/ins/boussinesq/benchmark.1e6 ................................................ [HEAVY] SKIP
test:finite_element/ins/boussinesq/benchmark.1e4 ................................................ [HEAVY] SKIP
test:finite_element/ins/boussinesq/benchmark.1e3 ................................................ [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow/cylindrical.rc .......................................... [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow/cylindrical.rc-with-temp ................................ [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow/rotated.plane-poiseuille-average ........................ [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow/rotated.plane-poiseuille-average-first .................. [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow/rotated.plane-poiseuille-rc ............................. [HEAVY] SKIP
test:finite_volume/ins/mms/channel-flow/rotated.plane-poiseuille-rc-first ....................... [HEAVY] SKIP
test:finite_volume/pwcns/channel-flow.transient-jac ....................................... [METHOD!=OPT] SKIP
test:finite_volume/ins/mms/skew-correction.vortex-skewness-corrected-action ..................... [HEAVY] SKIP
test:finite_volume/ins/mms/skew-correction.vortex-skewness-corrected ............................ [HEAVY] SKIP
test:finite_volume/ins/mixing_length_total_viscosity_material.capped_mixing_length/transient .... [HEAVY] SKIP
test:finite_element/ins/lid_driven/mixed-transient-steady.jac ............................. [METHOD!=OPT] SKIP
test:finite_volume/ins/mixing_length_eddy_viscosity_aux.capped_mixing_length/transient .......... [HEAVY] SKIP
test:finite_volume/ins/wall_distance_capped_mixing_length_aux.capped_mixing_length .............. [HEAVY] SKIP
test:finite_volume/ins/mms/cylindrical.2d-rc-diri ............................................... [HEAVY] SKIP
test:finite_volume/ins/action/errors.momentum-inlet-error ................................................. OK
```

---

Haha you are right. So if we are doing those extensions, and I agree we should, you might as well add all of them
```suggestion
*-debug
*-optimized
*-profiling
*-pro
*-oprofile
```

---

Ok Ill amend the commit and re-push

---

done